### PR TITLE
Fix default postsubmit.MaxConcurrency = 1 for promotion jobs

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://coreos.slack.com/archives/GB7NB0CUC/p1588003186030600?thread_ts=1588002752.029100&cid=GB7NB0CUC

/cc @stevekuznetsov @alvaroaleman 

Breaking change ... Need to prepare the PR for release change.
Fix tests if breaking.

Update:
Added unit tests and https://github.com/openshift/release/pull/8610

### CODE REVIEW

**Comment by alvaroaleman** [comment]
After we already failed once to implement this correctly, we should definitely add tests for this

File: pkg/jobconfig/files.go [hunk 1]
```diff
@@ -414,7 +414,9 @@ func mergePresubmits(old, new *prowconfig.Presubmit) prowconfig.Presubmit {
 	merged.AlwaysRun = old.AlwaysRun
 	merged.RunIfChanged = old.RunIfChanged
 	merged.Optional = old.Optional
-	merged.MaxConcurrency = old.MaxConcurrency
+	if old.MaxConcurrency != 0 {
```
Comment (alvaroaleman) [inline pkg/jobconfig/files.go:417]: Sorry, I don't understand this. We need the setting only for postsubmits and postsubmits must have this set to `1` as we only have build jobs as postsubmits. Why the check?

Comment (hongkailiu) [reply pkg/jobconfig/files.go:417]: Is this up to date?
https://github.com/openshift/ci-tools/blob/master/GENERATOR.md#hand-edited-prow-configuration

It did mention both presubmits and postsubmits.
However, periodic is not there.

Our prow-gen sets 1 only for the promotion job. We could have other postsubmits, right?

https://github.com/openshift/ci-tools/blob/9f3e703ba3e72ad422e61e105b95de682cb092a1/pkg/prowgen/prowgen.go#L175-L180

**Comment by hongkailiu** [comment]
After https://github.com/openshift/release/pull/8610 merges, this one should not be breaking any more.

File: pkg/jobconfig/files.go (hunk 1 above)
Comment (alvaroaleman) [reply pkg/jobconfig/files.go:417]: Right, we could have other postsubmits but I think there is no reason why a postsubmit should ever not run with max_concurrency: 1.

Comment (hongkailiu) [reply pkg/jobconfig/files.go:417]: It is the case for the current jobs.
This returns nothing.
> grep -irn max_ ./ci-operator/jobs | grep -v presubmits.yaml | grep -v "max_concurrency: 1"

Let me do it.

**Comment by hongkailiu** [comment]
/test breaking-changes

```
ll ci-operator/jobs/redhat-operator-ecosystem/playground/redhat-operator-ecosystem-playground-master-postsubmits.yaml
ls: ci-operator/jobs/redhat-operator-ecosystem/playground/redhat-operator-ecosystem-playground-master-postsubmits.yaml: No such file or directory
```

**Comment by hongkailiu** [comment]
I need to change this too.
https://github.com/redhat-operator-ecosystem/playground

File: pkg/jobconfig/files.go [hunk 2]
```diff
@@ -443,7 +447,9 @@ func mergePostsubmits(old, new *prowconfig.Postsubmit) prowconfig.Postsubmit {
 func mergePeriodics(old, new *prowconfig.Periodic) prowconfig.Periodic {
 	merged := *new
 
-	merged.MaxConcurrency = old.MaxConcurrency
+	if old.MaxConcurrency != 0 {
```
Comment (stevekuznetsov) [inline pkg/jobconfig/files.go:450]: Technically this means that if someone tries to set their job to have no limit on concurrency (which is what `0` means) we will stop them. I think I'd rather check the labels to see if it's a publish job?

Comment (hongkailiu) [reply pkg/jobconfig/files.go]: Done.

File: pkg/jobconfig/files.go (hunk 1 above)
Comment (hongkailiu) [reply pkg/jobconfig/files.go]: Not doing as discussed over slack.

**Comment by hongkailiu** [comment]
@alvaroaleman 

No breaking any more.

**Pull request merged**